wallet/src/main/java/com/android/identity_credential/wallet/presentation/OnlinePresentmentActivity.kt

```diff
@@ -0,0 +1,820 @@
+                    when (state) {
+                        State.PROCESSING -> {
+                            Logger.i(TAG, "State: Processing")
+                            stateDisplay.value = "Processing"
+                            processRequest(authorizationRequest, consentPromptData)
+                        }
+
+                        State.RESPONSE_SENT -> {
+                            Logger.i(TAG, "State: Response Sent")
+                            stateDisplay.value = "Response Sent"
+                        }
+
+                        else -> {}
+                    }
+                }
+
+                ScreenWithAppBar(title = "Presenting", navigationIcon = { }) {
+                    Column(
+                        modifier = Modifier
+                            .padding(16.dp),
+                        verticalArrangement = Arrangement.spacedBy(16.dp),
+                        horizontalAlignment = Alignment.CenterHorizontally
+                    ) {
+                        Text(
+                            text = "Sending mDL to reader.",
+                            style = MaterialTheme.typography.bodyMedium,
+                            color = MaterialTheme.colorScheme.onSurface
+                        )
+                        Text(
+                            text = "TODO: finalize UI",
+                            style = MaterialTheme.typography.bodyMedium,
+                            color = MaterialTheme.colorScheme.onSurface
+                        )
+                        HorizontalDivider()
+                        Text(
+                            text = "State: ${stateDisplay.value}",
+                            style = MaterialTheme.typography.bodyMedium,
+                            color = MaterialTheme.colorScheme.onSurface
+                        )
+                        HorizontalDivider()
+                        Button(onClick = { finish() }) {
+                            Text("Close")
+                        }
+                    }
+
+                    // when consent data is available, show consent prompt
+                    consentData = consentPromptData.value
+                    if (consentData != null) {
+                        ConsentPrompt(
+                            consentData = consentData!!,
+                            documentTypeRepository = walletApp.documentTypeRepository,
+                            onConfirm = {
+                                // create and send response on IO thread
+                                lifecycleScope.launch {
+                                    createAndSendResponse()
+                                }
+                            },
+                            onCancel = {
+                                finish()
+                            }
+                        )
+                    }
+                }
+            }
+        }
+    }
+
+    private fun processRequest(
```

Comment:
This whole function can likely be simplified once PR #631 and its follow up (which will use this logic in CredmanPresentationActivity and PresentationActivity) has landed.

Reply:
Let's land this PR first and Dritan's PR can follow.